### PR TITLE
(Port) Port Pixelorama to Ubuntu Touch

### PR DESCRIPTION
**Pull request by alicerunsonfedora**
This PR includes new assets and changes to add very basic support on [Ubuntu Touch (UT)](https://ubuntu-touch.io). A new `Misc/Clickable` directory exists that includes files needed to create a UT click package for both the armhf and arm64 architectures. There is a README in that directory as well, which outlines the build process.

Some settings have been changed for a custom feature `clickable`:
- First, the display orientation has been set to `sensor`, so that it rotates correctly
- HiDPI GUI is enabled

Additionally, a section in Main.gd has been added to account for `clickable`:

```gdscript
48	Global.open_sprites_dialog.current_dir = OS.get_system_dir(OS.SYSTEM_DIR_DESKTOP)
49	Global.save_sprites_dialog.current_dir = OS.get_system_dir(OS.SYSTEM_DIR_DESKTOP)
50
51	# FIXME: OS.get_system_dir does not grab the correct directory for Ubuntu Touch.
52	# Additionally, AppArmor policies prevent the app from writing to the /home
53	# directory. Until the proper AppArmor policies are determined to write to these
54	# files accordingly, use the user data folder where cache.ini is stored.
55	# Ubuntu Touch users can access these files in the File Manager at the directory
56	# ~/.local/pixelorama.orama-interactive/godot/app_userdata/Pixelorama.
57	if OS.has_feature("clickable"):
58		Global.open_sprites_dialog.current_dir = OS.get_user_data_dir()
59		Global.save_sprites_dialog.current_dir = OS.get_user_data_dir()
```

## Demo

Below is a screenshot of the app running at 2x scale on a Nexus 7 running UT in staged mode:
screenshot20210816_193124727

## Gotchas

These are some important notes for this port:

- The app seems to work best for tablet devices and not phones, though it can technically run.
- The current AppArmor policy in the click folder doesn't allow users to save to anywhere in `/home/phablet`. As a mitigation, the app will save to the user data folder.
- Currently, the GitHub workflows added to build these packages do _not_ upload to the [OpenStore](https://open-store.io). It is recommended that the owners or primary maintainers of the project create an account and publish the app there.
- There are some cases of unexpected crashes and freezes, though I am unsure if this is due to memory limitation only my particular device or something with the upstream codebase.
- The release workflows have _not_ been tested, but it should be similar enough to the dev CI script.
- The on-screen keyboard does not display when editing text. This may be an Ubuntu Touch problem, or the Godot port doesn't support it yet.

### CODE REVIEW

File: .github/workflows/dev-clickable-builds.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  pull_request:
+    branches: [ master ]
+    paths-ignore: 
+    - "Translations/*"
+    - "installer/*.pot"
+    - "installer/po/*"
+
+env:
+  GODOT_VERSION: 3.3
+  GODOT_VERSION_MAC: 3.3
```
Comment (OverloadedOrama) [inline .github/workflows/dev-clickable-builds.yml:14]: Since this workflow runs only on Ubuntu, I don't think that GODOT_VERSION_MAC is needed here.

Comment (alicerunsonfedora) [reply .github/workflows/dev-clickable-builds.yml:14]: Whoops, I didn't look at it closely enough. I'll remove it

**Review by OverloadedOrama — APPROVED** [review]
Sorry for the delay, I had been busy with v0.9 lately. I think it looks good, although I do not own an Ubuntu Touch device so I'm unable to test the port myself. I have created an account in the OpenStore and I started looking into publishing Pixelorama there. I'll let you know when this is done.

For now, I think this PR looks good to get merged. Thank you very much!

**Comment by jonnius** [comment]
FYI: Clickable has a Godot template now and ships a UT compatible version of Godot within its containers. You might want to have a look at it to simply these build scripts.

**Comment by OverloadedOrama** [comment]
I just realized that this PR changes some of the shortcuts, such as undo and redo, among others, and adds the `Meta` key in the combination. @alicerunsonfedora is this something that is required for Ubuntu Touch? Because we'd have to find another way of implementing this, as the way the PR changes the shortcuts affects all platforms, and I'll have to change that.

**Comment by alicerunsonfedora** [comment]
I don’t believe I touched the shortcuts at all.

**Comment by OverloadedOrama** [comment]
If you look at the changes in the project.godot file, there are indeed changes in some of the shortcuts. I suppose they were accidental, so I'll go ahead and revert them.

**Comment by OverloadedOrama** [comment]
Any idea why the `dev-clickable-builds` checks are failing? They seem to be failing for all commits after I published Pixelorama to the Open Store.

**Comment by alicerunsonfedora** [comment]
> Any idea why the `dev-clickable-builds` checks are failing? They seem to be failing for all commits after I published Pixelorama to the Open Store.
> 
> EDIT: It's possible that the problem lies in [nodesource/distributions#1271](https://github.com/nodesource/distributions/issues/1271) and [nodesource/distributions#1266](https://github.com/nodesource/distributions/issues/1266), hopefully they will fix it soon.

It appears to be that way. I wonder if development builds for other UT apps are failing because of this issue, though the workflows themselves should be operating fine and have no relation to publishing on the OpenStore.

**Pull request merged**